engine/chain/errors/bhs_error_response.go

```diff
@@ -0,0 +1,57 @@
+package chainerrors
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	chainmodels "github.com/bitcoin-sv/spv-wallet/engine/chain/models"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/go-resty/resty/v2"
+)
+
+// MapBHSErrorResponseToSpverror is a method that will check what kind of response came back from
+// Block Header Service and map it to spverror and set it to context
+func MapBHSErrorResponseToSpverror(res *resty.Response) models.SPVError {
```

Comment:
Why this is public, does it need to be in package errors? for me it's rather a part of the logic of the service

Reply:
I made it public so it can be imported in bhs service, and put it in error package, because it seemed like a good place to handle error responses.  It is indeed only used in a service so I will move it

Reply:
moved